pkg/apis/core/types_shoot.go

```diff
@@ -88,6 +88,8 @@ type ShootSpec struct {
 	Resources []NamedResourceReference
 	// Tolerations contains the tolerations for taints on seed clusters.
 	Tolerations []Toleration
+	// ManagedSeedName is the name of the ManagedSeed resource for this Shoot.
+	ManagedSeedName *string
```

Comment:
It's needed because there are currently settings in the `use-as-seed` annotation for the shoot API server. They are actually part of the new API as well, see `spec.shoot.apiServer`, to maintain compatibility and because they might actually be important I guess. To actually apply these settings, the botanist (in the Shoot controller) needs to get hold of the managed seed in order to read them, similarly to how it read the `use-as-seed` annotation before. Also, this is intentionally part of the spec and not e.g. an annotation, because we want to trigger a reconciliation every time it changes, in order to apply the new settings.

Reply:
See my comment above: How is this field triggering reconciliation for an existing `Shoot` and `ManagedSeed` where setting like the API server replicas are changed?

Reply:
Yes, you are right, and also for this reason I agree with @timebertt that this mechanism should be reworked entirely, see https://github.com/gardener/gardener/pull/3177#discussion_r544344146.